root/programs/C#/Frameworks/Infrastructure/ServiceInterface/AsyncProcessingService/AsyncProcessingService.cs

```diff
@@ -0,0 +1,222 @@
+        {
+            _asyncParameterValue = new AsyncProcessingServiceParameterValue("AsyncProcessingService", "Update", "Update", "SQL",
+                                                                             new MyUserInfo("AsyncProcessingService", "AsyncProcessingService"));
+            _asyncParameterValue.UserId = userId;
+            _asyncParameterValue.ExecutionStartDateTime = executionStartDateTime;
+            _asyncParameterValue.RegistrationDateTime = registrationDateTime;
+
+            // Based on the service status setting database command value
+            switch (checkStatus)
+            {
+                case AsyncProcessingServiceParameterValue.AsyncStatus.Processing:
+                    _asyncParameterValue.ProgressRate = 50;
+                    _asyncParameterValue.StatusId = (int)AsyncProcessingServiceParameterValue.AsyncStatus.Processing;
+                    _asyncParameterValue.CommandId = (int)AsyncProcessingServiceParameterValue.AsyncCommand.Stop;
+                    break;
+                case AsyncProcessingServiceParameterValue.AsyncStatus.Stop:
+                    _asyncParameterValue.ProgressRate = 100;
+                    _asyncParameterValue.StatusId = (int)AsyncProcessingServiceParameterValue.AsyncStatus.Stop;
+                    _asyncParameterValue.CommandId = 0;
+                    break;
+                case AsyncProcessingServiceParameterValue.AsyncStatus.AbnormalEnd:
+                    _asyncParameterValue.ProgressRate = 0;
+                    _asyncParameterValue.StatusId = (int)AsyncProcessingServiceParameterValue.AsyncStatus.AbnormalEnd;
+                    _asyncParameterValue.CommandId = 0;
+                    break;
+                case AsyncProcessingServiceParameterValue.AsyncStatus.Abort:
+                    _asyncParameterValue.ProgressRate = 0;
+                    _asyncParameterValue.StatusId = (int)AsyncProcessingServiceParameterValue.AsyncStatus.Abort;
+                    _asyncParameterValue.CommandId = 0;
+                    break;
+            }
+            _asyncParameterValue.NumberOfRetries = numberOfRetries;
+            _asyncParameterValue.CompletionDateTime = completionDateTime;
+
+            return _asyncParameterValue;
+        }
+
+        /// <summary>
+        /// Set things in motion so your service can do its work.
+        /// </summary>
+        protected override void OnStart(string[] args)
+        {
+            StartByThreadPool(AsyncProcessingServiceParameterValue.AsyncStatus.Processing);
+        }
+
+        /// <summary>
+        /// Calls callback method using Threadpool worker item.
+        /// </summary>
+        /// <param name="check"></param>
+        /// <returns></returns>
+        public bool StartByThreadPool(AsyncProcessingServiceParameterValue.AsyncStatus check)
+        {
+            int maxThreadCount;
+
+            lock (this)
```

Comment:
Here is locked, but I think that this section does not work in a multi-threaded in this implementation.